### PR TITLE
feat: add new CRD `greptimedbstandalone`

### PR DESCRIPTION
**Pull request by zyy17**
## What's changes

Add the new CRD `greptimedbstandalone` to maintain the standalone greptimedb:

```shell
cat <<EOF | kubectl apply -f -
apiVersion: greptime.io/v1alpha1
kind: GreptimeDBStandalone
metadata:
  name: basic
  namespace: default
spec:
  base:
    main:
      image: greptime/greptimedb:latest
EOF
```

It will create the standalone greptimedb.

Related issue: #125 .

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -122,6 +122,30 @@ rules:
...
+  - update
+  - watch
+- apiGroups:
+  - greptime.io
+  resources:
+  - greptimedbstandalones/finalizers
+  verbs:
+  - update
+- apiGroups:
+  - greptime.ioresources=greptimedbstandalones/status
```
Comment (daviderli614) [inline config/rbac/role.yaml:144]: suggestion:
```
- apiGroups:
  - greptime.io
  resources:
  - greptimedbstandalones/status 
```

File: cmd/operator/app/command.go [hunk 2]
```diff
@@ -90,6 +91,11 @@ func NewOperatorCommand() *cobra.Command {
 				os.Exit(1)
 			}
 
+			if err := greptimedbstandalone.Setup(mgr, o); err != nil {
+				setupLog.Error(err, "unable to setup controller", "controller", "greptimedbstandalone")
+				os.Exit(1)
+			}
+
```
Comment (daviderli614) [inline cmd/operator/app/command.go:98]: suggestion：Use greptimedb standalone need to install crd, default not setup?

File: controllers/greptimedbstandalone/controller.go [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+	}
+
+	// remove our finalizer from the list.
+	controllerutil.RemoveFinalizer(standalone, greptimedbStandaloneFinalizer)
+
+	if err := r.Update(ctx, standalone); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	klog.Infof("Delete GreptimeDB cluster '%s/%s'", standalone.Namespace, standalone.Name)
```
Comment (daviderli614) [inline controllers/greptimedbstandalone/controller.go:216]: suggestion: Delete GreptimeDB standalone

File: config/rbac/role.yaml (hunk 1 above)
Comment (zyy17) [reply config/rbac/role.yaml:144]: Typo in `controller.go` comments...

File: cmd/operator/app/command.go (hunk 2 above)
Comment (zyy17) [reply cmd/operator/app/command.go:98]: same as `greptimedbcluster`. you need to `make install` to install greptimedbstandalone.

**Review by daviderli614 — APPROVED** [review]
LGTM

**Pull request merged**